### DISCUSSION TITLE
Pipeline i trigger na master/main

### DISCUSSION THREAD

**Discussion by aiv** [AzureCommunityPL/group#339] [Pomoc]
Hej

Z jakiegoś dziwnego powodu nie działa mi trigger po merge PR do master/main.
W yamlu mam zadefiniowane:

```
trigger:
  - develop
  - main
```

w UI nie mam włączonego nadpisywania yamla. Pipeline uporczywie buduje mi każdy PR (ale condition w stages to wykluczają) a pomija master/main.

Jak włączyłem w UI nadpisywanie i ustawiłem develop, main i bez PR to przestało budować PR, ale maina dalej nie dotyka.

Jakiś pomysł?

**Comment by aryll** [comment]
Zastanawiam się czy to nie kwestia formatowania?
Ja mam tak i "u mnie działa"

![image](https://user-images.githubusercontent.com/5950108/131634823-a8751042-a618-4efa-ad70-134bbc466f9c.png)

**Reply by aiv** [reply]
Tutaj to niestety bez znaczenia. https://jsonformatter.org/yaml-parser
Bez względu czy spacja będzie czy nie, powstanie z tego tablica dwuelementowa :(

**Comment by kamil-mrzyglod** [comment]
Ja pomysłu nie mam, ale zrobiłem vanilla pipeline i bangla.

YAML:
```
trigger:
- master
- develop

pool:
  vmImage: windows-latest

steps:

- script: |
    az version
  displayName: 'Run a multi-line script'
```
PR feature branch do develop - CI triggerowane.
PR develop do master - CI triggerowane.

Próbowałem dać jaki biały znak przy definiowaniu listy - działa.

Sugerowałbym:
- upewnij się, że masz `main` a nie `master`
- postaw vanilla pipeline i zobacz, czy on się uruchomi prawidłowo
- sprawdź, czy przypadkiem na masterze / mainie nie masz już takiego commita i stąd brak triggera

**Reply by aiv** [reply]
ale to trigeruje się PR czy już po PR jak wejdzie na main?
u mnie PR działał (chodź tego nie chciałem) a nie działało już dodanie kodu na main (w efekcie PR)

**Reply by kamil-mrzyglod** [reply]
To co się triggeruje to branch `develop` / `master` jak zrobisz merge z innego brancha i push do repo. Jak chcesz triggera w PR jak jest review to możesz sobie ustawić build policy i tam skonfigurować co trzeba. Wtedy nie ustawiasz triggerów w YAMLu.

**Reply by aiv** [reply]
no właśnie nie chcę PR a samo margowanie z develop do master.
Sprawdze one more time

**Reply by aiv** [reply]
Sprawdziłem i na samplu działa, a na produkcyjnem pipeline już nie :)
Jakiś pomysł na debugowanie trigerrów?

**Reply by kamil-mrzyglod** [reply]
Możesz odpalić z zmienną systemową Debug - https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemdebug i być może coś więcej tam wyjdzie ale proponowałbym też starą szkołę - przeklej z sampla konfigurację triggerów i zobacz czy to nie załatwia sprawy.

**Reply by kamil-mrzyglod** [reply]
Ewentualnie w drugą stronę - do sampla przeklej resztę swojego pipeline'u.

**Comment by kmadof** [comment]
Dalej masz ten problem?

Ten master Ci się odpala, ale nie idą joby/stage'a? Czy w ogóle Ci nie startuje?

**Reply by aiv** [reply]
tak, mam wrażenie że to ustawienia organizacji/repo na GH mogą być winne.
na prywatnym repo all działa, na firmowym już nie (tylko devel, main nie)

**Reply by kmadof** [reply]
Z ciekawości wyrzuć całkowcie sekcję `triggers` wtedy pójdzie domyślnie, a to znaczy że powinien działać na każdy commit.

**Reply by aiv** [reply]
Dobry kierunek, thx sprawdzę i dam znać
